src/commands/periscope/periscope.ts

```diff
@@ -0,0 +1,163 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from 'vscode-azureextensionui';
+import * as tmpfile from '../utils/tempfile';
+import * as clusters from '../utils/clusters';
+import { getExtensionPath, longRunning } from '../utils/host';
+import {
+    getClusterDiagnosticSettings,
+    chooseStorageAccount,
+    getStorageInfo,
+    prepareAKSPeriscopeDeploymetFile,
+    generateDownloadableLinks,
+    getWebviewContent
+} from './helpers/periscopehelper';
+import { PeriscopeStorage } from './models/storage';
+import AksClusterTreeItem from '../../tree/aksClusterTreeItem';
+import { reporter } from '../utils/reporter';
+
+export default async function periscope(
+    context: IActionContext,
+    target: any
+): Promise<void> {
+    const kubectl = await k8s.extension.kubectl.v1;
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    // ToDo: Generalise this for other feature as wrapper.
+    reporter.sendTelemetryEvent("command", { command: "aks.periscope" });
+
+    if (cloudExplorer.available && kubectl.available) {
+        const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+
+        if (clusterTarget && clusterTarget.cloudName === "Azure" &&
+            clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster") {
+            const cluster = clusterTarget.cloudResource as AksClusterTreeItem;
+            const clusterKubeConfig = await clusters.getKubeconfigYaml(cluster);
+
+            if (clusterKubeConfig) {
+                await runAKSPeriscope(cluster, clusterKubeConfig);
+            }
+        } else {
+            vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+        }
+    }
+}
+
+async function runAKSPeriscope(
+    cluster: AksClusterTreeItem,
+    clusterKubeConfig: string
+): Promise<void | undefined> {
+    const clusterName = cluster.name;
+
+    // Get Diagnostic settings for cluster and get associated storage account information.
+    const clusterDiagnosticSettings = await longRunning(`Identifying cluster diagnostic settings.`,
+        () => getClusterDiagnosticSettings(cluster)
+    );
+
+    if (!clusterDiagnosticSettings || !clusterDiagnosticSettings.value?.length) {
+        // If there is no storage account attached to diagnostic setting, don't move forward and at this point we will render webview with helpful content.
+        await createPeriscopeWebView(cluster.name, undefined, undefined, false);
+        return undefined;
+
+    }
+
+    const clusterStorageAccountId = await longRunning(`Identifying cluster associated storage account.`,
+        () => chooseStorageAccount(clusterDiagnosticSettings)
```

Comment:
This can display interactive UI so shouldn't have a `longRunning` around it.  I think that function is quick on the non-interactive paths so you can probably just remove the `longRunning` - if there is nothing to select, then you don't need the progress popup, and if there is, then the quick pick list shows the user that you are waiting for _them_ (rather than for a background process, which is what the progress popup is for).